src/components/TargetMarker/index.js

```diff
@@ -0,0 +1,36 @@
+import React from 'react';
+import { useSelector } from 'react-redux';
+
+import { Marker } from 'react-native-maps';
+import strings from 'localization';
+import { number } from 'prop-types';
+import TopicIcon from 'components/TopicIcon';
+
+const TargetMarker = ({ latitude, longitude, topicId }) => {
+  const { topics } = useSelector(({ topics }) => ({
+    topics: topics?.topics ?? [],
+  }));
+
+  const { topic } = topics?.find(({ topic: { id } }) => id === topicId) || {};
+
+  if (!topic) return null;
```

Comment:
```suggestion
  if (!topic?.id) return null;
```
I see that "topic" is an object

Reply:
It was just to prevent an error when you enter to the app and topic is `undefined`. Because, after that, a topic will always be there. But you're right, check for `topic.id `is a better implementation. Great!

Reply:
It's always false except when topic is `undefined` and in that case `!undefined` return true